src/utils.jl

```diff
@@ -9,7 +9,7 @@ end
 function serialize_proto(data)
     pb = PipeBuffer()
     encode(ProtoEncoder(pb), data)
-    pb.data
+    return view(pb.data, 1:pb.size) # Do not return additional zeros
```

Comment:
@JamieMair 
This seems like the core of this PR (and the other changes are for reuse of this in other places).
But I can't understand the difference between this and the previous implementation.
Can you give a short explanation and links?

Overall the PR looks good I just want to do some due diligence

Reply:
Under the hood, `ProtoBuf.jl` uses the function `maybe_ensure_room` ([link](https://github.com/JuliaIO/ProtoBuf.jl/blob/548017cb69778cb5f0cd1273b16507f81d64719e/src/codec/encode.jl#L7C2-L7C2)), which expands an IO buffer to potentially hold more bytes (padding zeros usually), so that new bytes can be written without unnecessary resizing. Sometimes this guesses the amount of bytes to expand by and overshoots (as with the example in https://github.com/JuliaLogging/TensorBoardLogger.jl/issues/131), but the padding bytes should not be written to the events file, as it causes parsing errors. 

In https://github.com/JuliaLogging/TensorBoardLogger.jl/issues/131, the size of an event wrote the size of the unpadded bytes to the file, but the when skipping past the actual payload, ran into the padded zero bytes and tried to read this as the checksum, which caused the crc error. Also the checksum should only be ran on the unpadded data.

I actually ran into this problem with the hparams implementation but didn't really understand why `serialize_proto` wouldn't work for my application, which is why I wrote the function with `take!`. Now that I know why it wasn't working, I figured I should also update the hparams implementation to use the updated `serialize_proto` function instead. The main difference between this and my old function is the `view` which avoids the extra allocation.

Reply:
Lets change back to the `take!` version.
It looks like it does not allocate and we would avoid possible future breaking changes.

note:
I tested `take!` using
```julia
julia> using BenchmarkTools
julia> pb = PipeBuffer()
julia> write(pb, rand(1000))
julia> @benchmark take!(pb)
Time  (median):      37.376 ns
Memory estimate: 64 bytes, allocs estimate: 1.

julia> write(pb, rand(1000000))
julia> @benchmark take!(pb)
Time  (median):      36.753 ns
Memory estimate: 64 bytes, allocs estimate: 1.
```

Reply:
```suggestion
    return take!(pb) # Do not return additional zeros
```